### PR TITLE
HW-52031. Update select method presenter

### PR DESCRIPTION
**Pull request by azakrevska-epam**

### CODE REVIEW

File: ui/src/main/java/com/hyperwallet/android/ui/transfermethod/SelectTransferMethodPresenter.java [hunk 1]
```diff
@@ -35,14 +39,17 @@
 public class SelectTransferMethodPresenter implements SelectTransferMethodContract.Presenter {
     private static final String TAG = SelectTransferMethodPresenter.class.getName();
 
-    private static final String INDIVIDUAL = "INDIVIDUAL";
     private final TransferMethodConfigurationRepository mTransferMethodConfigurationRepository;
+    private final UserRepository mUserRepository;
     private final SelectTransferMethodContract.View mView;
+    private HyperwalletUser mUser;
```
Comment (fmattos-hw) [inline ui/src/main/java/com/hyperwallet/android/ui/transfermethod/SelectTransferMethodPresenter.java]: We don't need to keep a reference to the HyperwalletUser - we're already doing it in the repository. With that said, please revert all those "build" callback methods. Most of that refactoring is actually not necessary as we only need the profile type (nesting the get user call) in some specific cases

File: ui/src/main/java/com/hyperwallet/android/ui/transfermethod/SelectTransferMethodPresenter.java [hunk 2]
```diff
@@ -91,75 +80,135 @@ public void loadCurrency(final boolean forceUpdate, @NonNull final String countr
...
+        if (mUser == null) {
+            final UserRepository.LoadUserCallback loadUserCallback = buildLoadUserCallback(loadMethodTypesKeysCallback);
+            mUserRepository.loadUser(loadUserCallback);
+        } else {
+            mTransferMethodConfigurationRepository.getKeys(loadMethodTypesKeysCallback);
+        }
+    }
+
+    @Override
+    public void openAddTransferMethod(@NonNull final String country, @NonNull final String currency,
```
Comment (fmattos-hw) [inline ui/src/main/java/com/hyperwallet/android/ui/transfermethod/SelectTransferMethodPresenter.java]: This method should take one extra parameter, the profile type and in AddTransferMethodActivity add a new extra,
EXTRA_TRANSFER_METHOD_PROFILE_TYPE which will be used in AddTransferMethodPresenter#loadTransferMethodConfigurationFields. See https://confluence.site1.hyperwallet.local/display/PF/Technical+Design+-+Add+Transfer+Method
My apologies, I forgot to link the design to the ticket

File: ui/src/main/java/com/hyperwallet/android/ui/transfermethod/SelectTransferMethodPresenter.java (hunk 1 above)
Comment (azakrevska-epam) [reply ui/src/main/java/com/hyperwallet/android/ui/transfermethod/SelectTransferMethodPresenter.java]: @fmattos-hw Done

File: ui/src/main/java/com/hyperwallet/android/ui/transfermethod/SelectTransferMethodPresenter.java (hunk 2 above)
Comment (azakrevska-epam) [reply ui/src/main/java/com/hyperwallet/android/ui/transfermethod/SelectTransferMethodPresenter.java]: @fmattos-hw Done (see commit a8ace3d07496aef5157c990a74e76b10a157335a)

File: ui/src/test/resources/user_response.json [hunk 3]
```diff
@@ -0,0 +1,26 @@
+{
+  "token": "usr-f9154016-94e8-4686-a840-075688ac07b5",
+  "status": "PRE_ACTIVATED",
+  "createdOn": "2017-10-30T22:15:45",
+  "clientUserId": "CSK7b8Ffch",
```
Comment (fmattos-hw) [inline ui/src/test/resources/user_response.json]: can we use some other value? like 123456 ?

File: ui/src/test/java/com/hyperwallet/android/ui/repository/UserRepositoryImplTest.java [hunk 4]
```diff
@@ -103,23 +104,23 @@ public Object answer(InvocationOnMock invocation) {
...
-        assertThat(user.getField(HyperwalletUser.UserFields.DATE_OF_BIRTH), is("1991-01-01"));
-        assertThat(user.getField(HyperwalletUser.UserFields.EMAIL), is("testUser@hyperwallet.com"));
-        assertThat(user.getField(HyperwalletUser.UserFields.ADDRESS_LINE_1), is("575 Market Street"));
-        assertThat(user.getField(HyperwalletUser.UserFields.CITY), is("San Francisco"));
-        assertThat(user.getField(HyperwalletUser.UserFields.STATE_PROVINCE), is("CA"));
-        assertThat(user.getField(HyperwalletUser.UserFields.COUNTRY), is("US"));
-        assertThat(user.getField(HyperwalletUser.UserFields.POSTAL_CODE), is("94105"));
-        assertThat(user.getField(HyperwalletUser.UserFields.LANGUAGE), is("en"));
-        assertThat(user.getField(HyperwalletUser.UserFields.PROGRAM_TOKEN),
+        assertThat(resultUser.getField(TOKEN), is("usr-f9154016-94e8-4686-a840-075688ac07b5"));
```
Comment (fmattos-hw) [inline ui/src/test/java/com/hyperwallet/android/ui/repository/UserRepositoryImplTest.java]: maybe would be better to use the getters i.e getStatus, getCreatedOn etc

Comment (azakrevska-epam) [reply ui/src/test/java/com/hyperwallet/android/ui/repository/UserRepositoryImplTest.java]: @fmattos-hw Fixed in HW-52030

File: ui/src/test/resources/user_response.json (hunk 3 above)
Comment (azakrevska-epam) [reply ui/src/test/resources/user_response.json]: @fmattos-hw Missed. Done.

**Pull request merged**